modAionImpl/src/org/aion/zero/impl/sync/FastSyncManager.java

```diff
@@ -135,4 +220,107 @@ public void updateRequests(
             ensureCompleteness();
         }
     }
+
+    /**
+     * Processes a block response by checking the proof-of-work. Adds valid blocks to the import
+     * queue.
+     *
+     * @param peerId the numerical identifier of the peer who sent the response
+     * @param displayId the display identifier of the peer who sent the response
+     * @param response the response with blocks to be processed
+     */
+    public void validateAndAddBlocks(int peerId, String displayId, ResponseBlocks response) {
+        if (!executors.isShutdown()) {
+            executors.submit(
+                    new TaskValidateAndAddBlocks(
+                            peerId,
+                            displayId,
+                            response,
+                            blockHeaderValidator,
+                            downloadedBlocks,
+                            importedBlockHashes,
+                            receivedBlockHashes,
+                            log));
+        }
+    }
+
+    public void addToImportedBlocks(ByteArrayWrapper hash) {
+        this.importedBlockHashes.put(hash, null); // TODO: is there something useful I can add?
```

Comment:
what is the second arg supposed to be?

Reply:
FastSyncManager.java should be reviewed only in it's dedicated PR.